builder/azure/chroot/diskattacher.go

```diff
@@ -38,7 +37,10 @@ func (da *diskAttacher) DetachDisk(ctx context.Context, diskID string) error {
 	log.Println("Fetching list of disks currently attached to VM")
 	currentDisks, err := da.getDisks(ctx)
 	if err != nil {
-		return err
+		currentDisks, err = da.getScaleSetDisks(ctx)
+		if err != nil {
+			return err
+		}
```

Comment:
Is there a way to identify the exact error that can be ignored from getDisk? 

With this implementation we will lose the original error from calling da.getDisks(). It would be great to have both errors in the case that both function calls fail.

Reply:
Apologies for the lack of activity but I was quite stretched this week.

I agree that we should bubble the error up somehow for the user to be aware but I'm struggling to find a proper solution. With the current implementation, any error (or MultiError for that matter) which is returned from DetachDisk, AttachDisk and the likes will result in a `multistep.ActionHalt` in the main "step" which will essentially stop the build process. So this might not be the perfect solution.

Is there a decent way to surface such message as a "warning"? I'm thinking of something much simpler which is just to notify the user that the initial call failed and display what the error was before the falling back to the VMSS logic.

Let me share an example for a bit better understanding. Something along the lines of:
```
currentDisks, err := da.getDisks(ctx)
if err != nil {
  log.Printf("DetachDisk.getDisks: error: %+v\nFalling back to the VM scale set implementation...", err)
  // We can even make this explicit in the UI
  ui.Say("Fetching VM instance disks failed. Enable debugging for more details. Falling back to the VM scale set implementation...")
  currentDisks, err = da.getScaleSetDisks(ctx)
  if err != nil {
    return err
  }
}

Reply:
@nywilken, please let me know what you think on the above ☝️